src/webgpu/shader/validation/expression/access/vector.spec.ts

```diff
@@ -394,27 +436,47 @@ g.test('abstract')
   .params(u =>
     u
       .combine('vector_width', [2, 3, 4] as const)
-      .combine('element_type', ['int', 'float'] as const)
+      .combine('abstract_type', ['int', 'float'] as const)
+      .combine('concrete_type', ['u32', 'i32', 'f32', 'f16'])
       .beginSubcases()
       .combine('case', keysOf(kAbstractCases))
   )
+  .beforeAllSubcases(t => {
+    if (t.params.concrete_type === 'f16') {
+      t.selectDeviceOrSkipTestCase('shader-f16');
+    }
+  })
   .fn(t => {
+    const enables = t.params.concrete_type === 'f16' ? 'enable f16;' : '';
     const c = kAbstractCases[t.params.case];
-    const elem = t.params.element_type === 'int' ? '0' : '0.0';
+    const elem = t.params.abstract_type === 'int' ? '0' : '0.0';
     const vec_str = `vec${t.params.vector_width}(${Array(t.params.vector_width)
       .fill(elem)
       .join(', ')})`;
 
-    const code = `
+    const conversion_type =
+      c.result_width !== undefined
+        ? c.result_width === 1
+          ? `${t.params.concrete_type}`
+          : `vec${c.result_width}<${t.params.concrete_type}>`
+        : '';
+    const conversion = c.result_width !== undefined ? `const c: ${conversion_type} = r;` : '';
+
+    const code = `${enables}
 struct S {
   i : i32,
 }
 
 @compute @workgroup_size(1)
 fn main() {
   ${c.wgsl.replace('V', vec_str)}
+  ${conversion}
 }
 `;
-    const pass = typeof c.ok === 'function' ? c.ok(t.params.vector_width) : c.ok;
+    const convertible =
```

Comment:
nit (ignorable):

You could use:
https://github.com/gpuweb/cts/blob/7245ccb04c0a07a8c2e372f1ea330c1288a11ce0/src/webgpu/util/conversion.ts#L2441

```ts
const convertable = isConvertable(Type[`abstract-${t.params.abstract_type}`], Type[t.params.concrete_type]);
```
or something.

Reply:
Done